unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/ECSRuntime/Components/ECSComponentImplementations/PooledComponentImplementation.cs

```diff
@@ -0,0 +1,39 @@
+using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Models;
+using System;
+using UnityEngine;
+
+public class PooledComponentImplementation<ModelType> : IECSComponentImplementation<ModelType>
+{
+    private readonly ECSComponent<ModelType> component;
+
+    public PooledComponentImplementation(ECSComponent<ModelType> component)
+    {
+        this.component = component;
+    }
+
+    public void SetModel(IParcelScene scene, long entityId, ModelType model)
+    {
+        if(model != null)
+        {
+            component.SetModel(scene, entityId, model);
```

Comment:
how does this work? looks like `ECSComponent::SetModel` calls `PooledComponentImplementation::SetModel` which is calling `ECSComponent::SetModel`
am I missing something? 🤔